internal/bzlmod/go_deps.bzl

```diff
@@ -85,57 +134,72 @@ def _go_deps_impl(module_ctx):
             # dependencies and can thus only report implicit version upgrades
             # for direct dependencies. For manually specified go_deps.module
             # tags, we always report version upgrades.
-            if getattr(module, "is_root", False) and getattr(module_tag, "direct", True):
+            if in_root_module and getattr(module_tag, "direct", True):
                 root_versions[module_tag.path] = raw_version
             version = semver.to_comparable(raw_version)
             if module_tag.path not in module_resolutions or version > module_resolutions[module_tag.path].version:
+                # See file "go_repository.bzl" for the default values' definition.
+                build_directives = []
+                override = gazelle_overrides.pop(module_tag.path, None)
+                if override:
+                    build_directives = override.directives
+                else:
+                    # Translate deprecated attributes for now.
+                    synthesized_override = False
```

Comment:
I implemented this suggestion, but when testing it, I discovered that this runs afoul of [_rules_go_ using it](https://github.com/bazelbuild/rules_go/blob/7ed6bdea61e682b57adf4684ec1f47d174e817e4/MODULE.bazel#L34) today, and [this module depends on _rules_go_](https://github.com/bazelbuild/bazel-gazelle/blob/39aa39511f59223b242453d60e4e3f6f89ebf204/MODULE.bazel#L9). That means that we run into an overriding "build_file_proto_mode" attribute on `go_deps.module` in a non-root module.

The diff for addressing this desire looks like this:
```diff
@@ -142,14 +142,18 @@ def _go_deps_impl(module_ctx):
                 if override:
                     build_directives = override.directives
                 else:
-                    # Translate deprecated attributes for now.
+                    # Translate deprecated attributes for now, but only in the root module.
                     synthesized_override = False
                     build_naming_convention_override = getattr(module_tag, "build_naming_convention", "")
                     if build_naming_convention_override:
+                        if not module.is_root:
+                            fail("Overriding the \"build_naming_convention\" attribute in a \"go_deps.module\" tag is forbidden in non-root modules, but is specified in module \"{}\".".format(module.name))
                         build_directives.append("gazelle:go_naming_convention {}".format(build_naming_convention_override))
                         synthesized_override = True
                     build_file_proto_mode_override = getattr(module_tag, "build_file_proto_mode", "")
                     if build_file_proto_mode_override:
+                        if not module.is_root:
+                            fail("Overriding the \"build_file_proto_mode\" attribute in a \"go_deps.module\" tag is forbidden in non-root modules, but is specified in module \"{}\".".format(module.name))
                         build_directives.append("gazelle:proto {}".format(build_file_proto_mode_override))
                     elif not synthesized_override:
                         build_directives = DEFAULT_DIRECTIVES_BY_PATH.get(module_tag.path, [])
```

Reply:
Let's carve out a temporary exception for rules_go, which is otherwise difficult to update due to the cyclic dependency with Gazelle. I'll check the registry for any other offending modules, but I don't think there are any.

This error message would be an ideal place for a call to contribute to the registry of default directives. Could you add a message such as "If you need this override for a module to be added to the Bazel Central Registry, file an issue at (new issue link) or submit a PR adding the required directives to  (link to directives.bzl)".

Reply:
I'll work on this in the coming day or two. I expect that we'll need several volleys of copyediting, but I don't mind.

Is there anything special I need to do to detect the _rules_go_ Bazel module by name? Is [the `bazel_module.name` field](https://bazel.build/rules/lib/bazel_module#name) trustworthy for this purpose?

Reply:
Yes, just allow this whenever the name field is `"rules_go"`.

Reply:
Please see the latest commit for my take on the more elaborate failure message that [you suggested](https://github.com/bazelbuild/bazel-gazelle/pull/1456/files#r1124312569).